OCP-Infrastructure/helm/scripts/functions/helm_uninstall.active

```diff
@@ -0,0 +1,128 @@
+    fi
+
+    if [[ "${#resources}" -eq 0 ]]; then
+        echo "no resources marked for deinstallation"
+        echo ""
+        usage_message
+        exit 1
+    fi
+
+    ## UNINSTALL
+    for item in ${REVERSE_RESOURCE_ORDER[@]}; do
+        if [[ " ${resources[@]} " =~ "${item}" ]]; then
+            # use correct namespace
+            if [[ "${item}" == "keycloak-commons" ]] || [[ "${item}" == "keycloak" ]]; then
+                namespace="${namespace_keycloak}"
+            else
+                namespace="${namespace_hogarama}"
+            fi
+            exec_command="helm uninstall helm-${item} -n ${namespace}"
+            eval remove-${item} "${exec_command}"
+
+            rc="$(echo $?)"
+            if [[ "${rc}" -gt 0 ]]; then
+                rc_overall=1
+                echo "Something went wrong trying to uninstall resource ${item} in namespace ${namespace}"
+            fi
+        fi
+    done
+
+    if [[ "${rc_overall}" -gt 0 ]];then
+        echo "There was an error trying to uninstall a provided resource. Please check previous log messages for more information"
+        exit 1
+    fi
+}
+# readonly definition of a function throws an error if another function
+# with the same name is defined a second time
+readonly -f helm-uninstall
+[ "$?" -eq "0" ] || return $?
+
+
+## SPECIAL FUNCTIONS TO HANDLE CUSTOM RESOURCES, SUBSCRIPTIONS, etc.
+function remove-default() {
+    command="${@}"
+    execute "${command}"
+}
+readonly -f remove-default
+[ "$?" -eq "0" ] || return $?
+
+function remove-keycloak-commons() {
+    local csvkeycloak=$(oc get subscription keycloak-operator -n ${namespace} -o yaml | grep currentCSV | cut -d ":" -f2 | awk '{$1=$1;print}')
+    remove-default "${@}"
+    execute "oc delete clusterserviceversion ${csvkeycloak} -n ${namespace}"
+}
+readonly -f remove-keycloak-commons
+[ "$?" -eq "0" ] || return $?
+
+function remove-keycloak() {
+    remove-default "${@}"
+}
+readonly -f remove-keycloak
+[ "$?" -eq "0" ] || return $?
+
+function remove-hogarama-commons() {
+    local csveap=$(oc get subscription eap-operator -n ${namespace} -o yaml | grep currentCSV | cut -d ":" -f2 | awk '{$1=$1;print}') && echo "found clusterserviceversion: ${csveap}"
+    local csvgrafana=$(oc get subscription grafana-operator -n ${namespace} -o yaml | grep currentCSV | cut -d ":" -f2 | awk '{$1=$1;print}') && echo "found clusterserviceversion: ${csvgrafana}"
+    local csvprometheus=$(oc get subscription prometheus-operator -n ${namespace} -o yaml | grep currentCSV | cut -d ":" -f2 | awk '{$1=$1;print}') && echo "found clusterserviceversion: ${csvprometheus}"
+    remove-default "${@}"
+    execute "oc delete clusterserviceversion ${csveap} -n ${namespace}"
+    execute "oc delete clusterserviceversion ${csvgrafana} -n ${namespace}"
+    execute "oc delete clusterserviceversion ${csvprometheus} -n ${namespace}"
+}
+readonly -f remove-hogarama-commons
+[ "$?" -eq "0" ] || return $?
+
+function remove-amq() {
```

Comment:
Code smell: general utility functions should not have dependencies to individual components. This breaks encapsulation of the component.